src/main/java/ledger/user_interface/ui_controllers/TransactionTableView.java

```diff
@@ -343,17 +346,20 @@ public void updateTransactionTableView() {
     private void handleDeleteTransactionFromTableView() {
         int selectedIndex = this.getSelectionModel().getSelectedIndex();
         if (selectedIndex >= 0) {
-            TransactionModel model = (TransactionModel) this.getItems().get(selectedIndex);
-            Transaction transactionToDelete = model.getTransaction();
+            ObservableList<TransactionModel> transactionModelsToDelete = this.getSelectionModel().getSelectedItems();
 
-            TaskWithArgs<Transaction> task = DbController.INSTANCE.deleteTransaction(transactionToDelete);
-            task.RegisterFailureEvent((e) -> {
-                updateTransactionTableView();
-                setupErrorPopup("Error deleting transaction.", e);
-            });
+            for (TransactionModel currentModel : transactionModelsToDelete) {
+                Transaction transactionToDelete = currentModel.getTransaction();
```

Comment:
The null pointer comes up on this line, and the object that is null is currentModel, I believe. When the exception is thrown, only one transaction will be deleted, and it'll be the last single transaction that was clicked (likely the end of the SHIFT-click range). All the other transactions will not be deleted.